src/PostgREST/Auth.hs

```diff
@@ -163,14 +164,25 @@ middleware appState app req respond = do
 -- | Used to retrieve and insert JWT to JWT Cache
 getJWTFromCache :: AppState -> ByteString -> Int -> IO (Either Error AuthResult) -> UTCTime -> IO (Either Error AuthResult)
 getJWTFromCache appState token maxLifetime parseJwt utc = do
-  checkCache <- C.lookup (getJwtCache appState) token
+
+  -- Call first to delete the expired JWTs
+  C.purgeExpired jwtCache
```

Comment:
I am wondering whether it is a good idea to add this in the hot path of request processing unless we are sure it is a constant time operation (and fast).

Otherwise it should be moved to a separate thread.